peps/pep-0733.rst

```diff
@@ -0,0 +1,670 @@
+mostly correspond to these themes, and each contains a combined
+description of the issues raised in that category, along with
+links to the individual issues. In addition, we included a section
+that aims to identify the different stakeholders of the C API,
+and the particular requirements that each of them has.
+
+
+C API Stakeholders
+==================
+
+As mentioned in the introduction, the C API was originally
+created as the internal interface between CPython's
+interpreter and the Python layer. It was later exposed as
+a way for third-party developers to extend and embed Python
+programs. Over the years, new types of stakeholders emerged,
+with different requirements and areas of focus. This section
+describes this complex state of affairs in terms of the
+actions that different stakeholders need to perform through
+the C API.
+
+Common Actions for All Stakeholders
+-----------------------------------
+
+There are actions which are generic, and required by
+all types of API users:
+
+* Define functions and call them
+* Define new types
+* Create instances of builtin and user-defined types
+  and perform operations on them
+* Introspect objects, including types, instances, and functions
+* Raise and handle exceptions
+* Import modules
+* Access to Python's OS interface
+
+The following sections look at the unique requirements of various stakeholders.
+
+Extension Writers
+-----------------
+
+Extension writers are the traditional users of the C API. Their requirements
+are the common actions listed above.
+
+Authors of Embedded Python Applications
+---------------------------------------
+
+Applications with an embedded Python interpreter. Examples are
+`Blender <https://docs.blender.org/api/current/info_overview.html>`__ and
+`OBS <https://obsproject.com/wiki/Getting-Started-With-OBS-Scripting>`__.
+
+They need to be able to:
+
+* Configure the interpreter (import paths, inittab, ``sys.argv``, etc.).
+* Interact with the execution model and program lifetime, including
+  clean interpreter shutdown and restart.
+* Represent complex data models in a way Python can use without
+  having to create deep copies.
+* Provide and import frozen modules.
+* Run and manage multiple independent interpreters (in particular, when
+  embedded in a library that wants to avoid global effects).
+
+Python Implementations
+----------------------
+
+Python implementations such as
+`CPython <https://www.python.org>`__,
+`PyPy <https://www.pypy.org>`__,
+`GraalPy <https://www.graalvm.org/python/>`__,
+`IronPython <https://ironpython.net>`__,
+`RustPython <https://github.com/RustPython/RustPython>`__,
+`MicroPython <https://micropython.org>`__,
+and `Jython <https://www.jython.org>`__), may take
+very different approaches for the implementation of
+different subsystems. They need:
+
+* The API to be abstract and hide implementation details.
+* A specification of the API, ideally with a test suite
+  that ensures compatibility.
```

Comment:
Agreed, but it's not important to make it at this stage in this document. It's definitely been discussed for future designs, though (in the capi-workgroup repositories).

Reply:
This document is about the Python C API. Not the CPython C API.

Reply:
Perhaps you should make that clear somewhere near the start of the document, since traditionally, the "Python C API" is being identified with whatever the CPython implementation exports (along with the different tiers that have popped up over the years).